cpp/src/arrow/array/builder_adaptive.cc

```diff
@@ -273,12 +283,16 @@ Status AdaptiveUIntBuilder::FinishInternal(std::shared_ptr<ArrayData>* out) {
   *out = ArrayData::Make(type(), length_, {null_bitmap, data_}, null_count_);
 
   data_ = nullptr;
-  capacity_ = length_ = null_count_ = 0;
+  capacity_ = length_ = committed_length_ = null_count_ = 0;
   return Status::OK();
 }
 
 Status AdaptiveUIntBuilder::AppendValuesInternal(const uint64_t* values, int64_t length,
                                                  const uint8_t* valid_bytes) {
+  // UnsafeAppendToBitmap expects length_ to be the pre-update value, satisfy it
+  DCHECK_EQ(length_, committed_length_ + pending_pos_);
```

Comment:
Could you delete committed_length_ by using `length_ -= pending_pos_` here?

Reply:
You're right, that works. Thank you!